### PR TITLE
Sony ARW Ljpeg tweaks

### PR DESCRIPTION
**Pull request by kmilos**
Follow-up to #484

### CODE REVIEW

File: src/librawspeed/decoders/ArwDecoder.cpp [hunk 1]
```diff
@@ -352,9 +357,7 @@ void ArwDecoder::DecodeLJpeg(const TiffIFD* raw) {
 
   PostProcessLJpeg();
 
-  const TiffEntry* size_entry = raw->hasEntry(TiffTag::SONYRAWIMAGESIZE)
-                                    ? raw->getEntry(TiffTag::SONYRAWIMAGESIZE)
-                                    : raw->getEntry(TiffTag::DEFAULTCROPSIZE);
+  const TiffEntry* size_entry = raw->getEntry(TiffTag::SONYRAWIMAGESIZE);
```
Comment (da-phil) [inline src/librawspeed/decoders/ArwDecoder.cpp:360]: so no fallback is necessary, we're always sure to have this EXIF marker available?

Comment (kmilos) [reply src/librawspeed/decoders/ArwDecoder.cpp:360]: > we're always sure to have this EXIF marker available?

Seems to be the case so far, as mentioned.

In the worst case, I'd rather return the uncropped image w/ black borders than the DNG crop, I feel it's more consistent with the stated behaviour:

> RawSpeed does NOT…
> 
> crop the image to the same sizes as manufactures, but supplies biggest possible images.

Comment (LebedevRI) [reply src/librawspeed/decoders/ArwDecoder.cpp:360]: FWIW, `getEntry()` throws if you query for an entry that will not be found.

**Comment by LebedevRI** [comment]
Makes sense, thank you!

**Comment by kmilos** [comment]
Do we feel cheeky enough to drop this on dt? ;)

I suspect they'll [freeze and tag 4.4 very soon](https://github.com/darktable-org/darktable/wiki/Releases-cycle)... @TurboGit

**Comment by LebedevRI** [comment]
Even though `LJpegDecompressor` is no longer affected, i'd still prefer to let it fuzz for a few more days.
What is the cut-off date?

**Comment by TurboGit** [comment]
@LebedevRI : It is planed for June 14th, I can move it one day on 15th if necessary.

**Comment by LebedevRI** [comment]
Ah, that's plenty of time then.
I'll wait some more, and update submodule on 13'th then.

**Pull request merged**